addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3523,25 +3523,24 @@ const SnippetOptionWidget = Widget.extend({
      * @param {SnippetOptionWidget|UserValueWidget} parentWidget
      * @returns {Promise}
      */
-    _renderXMLWidgets: function (parentEl, parentWidget) {
-        const proms = [...parentEl.children].map(el => {
+    _renderXMLWidgets: async function (parentEl, parentWidget) {
+        const proms = [...parentEl.children].map(async el => {
             const widgetName = el.tagName.toLowerCase();
-            if (!userValueWidgetsRegistry.hasOwnProperty(widgetName)) {
+            if (!Object.prototype.hasOwnProperty.call(userValueWidgetsRegistry, widgetName)) {
                 return this._renderXMLWidgets(el, parentWidget);
             }
 
             const infos = this._extraInfoFromDescriptionElement(el);
             const widget = registerUserValueWidget(widgetName, parentWidget || this, infos.title, infos.options, this.$target);
-            return widget.insertAfter(el).then(() => {
-                // Remove the original element afterwards as the insertion
-                // operation may move some of its inner content during
-                // widget start.
-                parentEl.removeChild(el);
-
-                if (widget.isContainer()) {
-                    return this._renderXMLWidgets(widget.el, widget);
-                }
-            });
+            await widget.insertAfter(el);
+            // Remove the original element afterwards as the insertion
+            // operation may move some of its inner content during
+            // widget start.
+            parentEl.removeChild(el);
+
+            if (widget.isContainer()) {
+                return this._renderXMLWidgets(widget.el, widget);
+            }
         });
         return Promise.all(proms);
```

Comment:
In fact I don't see what you are changing in this whole code... is not this equivalent? :thinking:

Reply:
As I was debugging at this level, I thought that maybe it was worth it to rewrite the code with an async function for easier reading, but maybe I should have mentioned this change in the commit message ? Or keep the PR minimalist and only include the fix ?

Reply:
Some people would tell you to never do that to avoid losing git history for "no reason". IMO:
- Never do it in stable unless you have to modify a lot of the code around
- If you do it master "without any reason", do it in a dedicated commit

Here, let's just not do it :wink: